keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+import PropTypes from 'prop-types';
+
+CharacterList.propTypes = {
+	userInfo: PropTypes.object.isRequired,
+	userPath: PropTypes.string.isRequired,
+};
+
+function CharacterList({ userInfo, userPath }) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState('');
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		setCharacters(userInfo.characters || {});
+		setCurrentCharacter(userInfo.currentCharacter);
+	}, [userInfo]);
+
+	const handleOpen = () => {
+		setOpen(true);
+	};
+
+	async function handleSubmit(fields) {
+		const { name, characterClass, spec, race } = fields;
+		let charactersSnapShot = await ref.child('/Characters');
+
+		const id = charactersSnapShot.push({
+			class: characterClass,
+			race,
+			name,
+			specs: {
+				[spec]: {
+					spec,
+					selected: true,
+					configured: false,
+				},
+			},
+		});
+		const idKey = id.key;
+		let snapshot = await ref.child(userPath);
+		let userCharacterData = await snapshot.once('value');
+
+		if (userCharacterData.exists()) {
+			snapshot
+				.child('characters')
+				.child(idKey) //name should probably be a real key
+				.set({ name: name });
+		} else {
+			snapshot.set({
+				currentCharacter: name,
+				characters: {
+					[idKey]: {
+						name,
+					},
+				},
+			});
+		}
+		setCharacters({ ...characters, idKey: { name: name } }); //immutabilityHelper here
+	}
+
+	return (
+		<React.Fragment>
+			{open && (
```

Comment:
I'm curious if the modal automatically hides if it's closed? the open && might not even be necessary